examples/model_selection/partial_auc.py

```diff
@@ -0,0 +1,163 @@
+mitigate the risk of missing fraud cases, we need to keep a low level of false
+negative rate (FNR). Please note that since `TPR = TP/P = (TP)/(TP + FN) = 1 -
+FNR` , the limitation on low FNR is equivalent to the limitation on high TPR.
+
+
+In :func:`sklearn.metrics.roc_auc_score`, with the options ``max_fpr`` and
+``min_tpr``, you could define the restrictions to ROC and get a standardized
+partial AUC value returned, which could be used as your model selection
+criteria.
+
+Let's visualize a ROC example to clarify the definitions. You could follow below
+examples and manually calculate standardized pAUC for better understanding.
+
+
+.. note::
+
+    See also :func:`sklearn.metrics.roc_auc_score`,
+             :ref:`sphx_glr_auto_examples_model_selection_plot_roc.py`
+
+"""
+# %%
+# Plot partial ROC area on one example
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn import metrics
+
+print(__doc__)
+
+# Author: Fengjun Wang
+# License: BSD 3 clause
+
+# Make a fake y_true and y_score
+y_true = np.array([0, 0, 1, 1])
+y_score = np.array([0.1, 0, 0.1, 0.01])
+max_fpr = 0.9
+min_tpr = 0.6
+
+# Calculate the fpr, tpr list
+fpr, tpr, thresholds = metrics.roc_curve(y_true=y_true, y_score=y_score, pos_label=1)
+auc = metrics.roc_auc_score(y_true=y_true, y_score=y_score)
+spauc = metrics.roc_auc_score(
+    y_true=y_true, y_score=y_score, max_fpr=max_fpr, min_tpr=min_tpr
+)
+print(f"The fpr list is: {fpr}")
+print(f"The tpr list is: {tpr}")
+print(f"By default, the ROC AUC is {auc:.3f}.")
+print(
+    f"With max_fpr {max_fpr} and min_tpr "
+    f"{min_tpr}, the standardized partial AUC is {spauc:.3f}."
+)
+plt.rcParams["figure.figsize"] = (10, 10)
+
+# Plot the min_tpr horizontal line
+if min_tpr is not None:
+    plt.axhline(y=min_tpr, color="g", linestyle="--", alpha=0.5, label="min_tpr")
+
+# Plot the max_fpr vertical line
+if max_fpr is not None:
+    plt.axvline(x=max_fpr, color="r", linestyle="--", alpha=0.5, label="max_fpr")
+
+plt.plot(fpr, tpr, color="darkorange", lw=2, label="ROC curve")
+
+# Plot the non-discriminant ROC line for reference:
+plt.plot(
+    [0, 1], [0, 1], color="navy", lw=2, linestyle="--", label="non-discriminant ROC"
+)
+
+plt.xlim([0.0, 1.0])
+plt.ylim([0.0, 1.05])
+plt.xlabel("False Positive Rate")
+plt.ylabel("True Positive Rate")
+plt.title("Receiver operating characteristic example")
+plt.legend(loc="lower right")
+plt.grid()
+plt.gca().set_aspect("equal", adjustable="box")
+
+if min_tpr is None:
+    min_tpr = 0.0
+if max_fpr is None:
+    max_fpr = 1.0
+# Fill the region which conforms to max_fpr with red:
+plt.fill_betweenx(
+    tpr,
+    x1=max_fpr,
+    x2=fpr,
+    where=(fpr <= max_fpr),
+    alpha=0.1,
+    interpolate=True,
+    color="red",
+)
+# Fill the region which conforms to min_tpr with green:
+plt.fill_between(
+    fpr,
+    y1=min_tpr,
+    y2=tpr,
+    where=(tpr >= min_tpr),
+    alpha=0.1,
+    interpolate=True,
+    color="green",
+)
+plt.show()
```

Comment:
This does not render in sphinx-gallery. I believe that the filename of the example needs to be renamed to `examples/model_selection/plot_partial_auc.py` for this to work.

Reply:
Actually we should probably merge this into the existing `plot_roc.py` as additional sections to avoid proliferation of example in the gallery which makes it hard to navigate.

Reply:
I renamed it. Can you review, and let me know do we want to merge it in `plot_roc`? `plot_roc` is a quite long section, maybe we just need an additional link to tell the user we have `partial auc`?

Reply:
I allowed black changes, since otherwise the Linting has errors